packages/core/src/data-grid/cells/boolean-cell.tsx

```diff
@@ -1,7 +1,31 @@
+import { assertNever } from "../../common/support";
 import { drawBoolean } from "../data-grid-lib";
 import { GridCellKind, BooleanCell } from "../data-grid-types";
 import { InternalCellRenderer } from "./cell-types";
 
+/**
+ * Checkbox behavior:
+ *
+ * true + click -> unchecked
+ * false + click -> checked
+ * indeterminate + click -> checked
+ * empty + click -> empty
```

Comment:
This behavior is achieved by simply dumping out a loadingCell instead. Why not have this act like a checkbox which is "empty" but when hovered it shows the empty state and when clicked becomes checked. This provides new behavior instead of reproducing already possible behavior.

Reply:
Huh, I didn't think about the loading cell. I agree, then.

I'll share this option with the issue's creator.